src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/UploadObligationsViewModelMap.cs

```diff
@@ -41,17 +45,22 @@ private void SetErrors(UploadObligationsViewModelMapTransfer source, UploadOblig
             var displayDataError = false;
             var displayFileError = false;
 
-            if (source.ErrorData != null)
+            if (source.DisplayNotification)
             {
-                displayDataError = source.ErrorData.Any(r => dataErrorTypes.Contains(r.ErrorType));
-                displayFileError =
-                    source.ErrorData.Any(r => r.ErrorType == SchemeObligationUploadErrorType.File);
-                model.NumberOfDataErrors = source.ErrorData.Count(r => dataErrorTypes.Contains(r.ErrorType));
-                model.DisplaySuccessMessage = displayDataError == false && displayFileError == false;
-            }
+                if (source.ErrorData != null)
+                {
+                    displayDataError = source.ErrorData.Any(r => dataErrorTypes.Contains(r.ErrorType));
```

Comment:
don't you need to filter on the data error only in here. at the moment, I think it will be true for any of the 3 types of SchemeObligationUploadErrorType enum

Reply:
sorry, my comments above is not correct, you have a list with the types to filter on, maybe just rename the variable to DataAndSchemeErrors?

Reply:
Will rename if its not clear, which variable?

Reply:
that one displayDataError, it checks for data and scheme errorrs?

Reply:
updated